### PR TITLE
Added Vault capabilities to fetch AWS creds

### PR DESCRIPTION
**Pull request by mvisonneau**

### CODE REVIEW

**Comment by paulRbr** [comment]
Thanks @mvisonneau for this!

Could you include the logic of the `vault.sh` file in an `elsif f [ -n "${VAULT_ADDR}" ]` condition of `terraform.sh#L28` please?

The goal of the Makefile is for you to have to have the least env variables to handle so I'd rather not add a task that "outputs" variables to export. It can do it automatically when you need them.

**Comment by mvisonneau** [comment]
Yep, absolutely 👍 I just rebased.

File: terraform.sh [hunk 1]
```diff
@@ -26,9 +27,29 @@ if (which pass >/dev/null 2>&1); then
     declare "${secret}"="${pass_secret}"
 fi
 
+if [ -n "${VAULT_ADDR}" ] && [ -z "${VAULT_TOKEN}" ]; then
```
Comment (paulRbr) [inline terraform.sh]: Could you explode the `if` in two parts?
```
if [ -n "${VAULT_ADDR}" ]; then
  if [ -z "${VAULT_TOKEN}" ]; then
    ...
  fi
  ...
fi
```

Comment (paulRbr) [reply terraform.sh]: In order to include all the logic you added below in the `if [ -n "${VAULT_ADDR}" ] && [ -n "${VAULT_TOKEN}" ]; then` branch *here*.

And declare the key and secrets (in the same way it's done with the `pass` password store binary) as such:
```
declare "${key}"="$(echo ${creds} | jq -r .access_key)"
declare "${secret}"="$(echo ${creds} | jq -r .secret_key)"
```

**Review by paulRbr — CHANGES_REQUESTED** [review]

File: terraform.sh (hunk 1 above)
Comment (mvisonneau) [reply terraform.sh]: 👍 sounds good

**Review by paulRbr — APPROVED** [review]

**Comment by paulRbr** [comment]
Looks good now! Thanks @mvisonneau

**Pull request merged**